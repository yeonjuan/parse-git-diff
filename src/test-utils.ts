import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
