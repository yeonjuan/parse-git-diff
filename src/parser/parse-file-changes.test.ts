import { createContext } from '../test-utils';
import parseFileChanges from './parse-file-changes';

describe('parseChunkHeader', () => {
  it('should parse normal chunk header.', () => {
    // prettier-ignore
    const src =
`diff --git a/.gitignore b/.gitignore
index dd87e2d..1d2a37f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 node_modules
 build
+coverage
diff --git a/README.md b/README.md
deleted file mode 100644
index 8229282..0000000
--- a/README.md
+++ /dev/null
@@ -1 +0,0 @@
-# parse-git-diff`;

    const result = parseFileChanges(createContext(src));

    expect(result).not.toBe(null);
    expect(result).toMatchSnapshot();
  });
});
