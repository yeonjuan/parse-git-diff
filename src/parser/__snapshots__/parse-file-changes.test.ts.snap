// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseChunkHeader should parse normal chunk header. 1`] = `
Array [
  Object {
    "chunks": Array [
      Object {
        "addedPos": Object {
          "lines": 3,
          "start": 1,
        },
        "changes": Array [
          Object {
            "content": "node_modules",
            "line": 6,
            "type": "Unchanged",
          },
          Object {
            "content": "build",
            "line": 7,
            "type": "Unchanged",
          },
          Object {
            "content": "coverage",
            "line": 8,
            "type": "Added",
          },
        ],
        "deletedPos": Object {
          "lines": 2,
          "start": 1,
        },
        "type": "Chunk",
      },
    ],
    "path": ".gitignore",
    "type": "ChangedFile",
  },
  Object {
    "chunks": Array [
      Object {
        "addedPos": Object {
          "lines": 0,
          "start": 0,
        },
        "changes": Array [
          Object {
            "content": "# parse-git-diff",
            "line": 15,
            "type": "Deleted",
          },
        ],
        "deletedPos": Object {
          "lines": 1,
          "start": 1,
        },
        "type": "Chunk",
      },
    ],
    "path": "README.md",
    "type": "DeletedFile",
  },
]
`;
